lib/recurly/Http.js

```diff
@@ -1,25 +1,69 @@
+const https = require('https')
+const casters = require('./Caster')
+
+// an adapter function to abstract away some of
+// the low level details of requests and responses in node
+function makeRequest (options, requestBody) {
```

Comment:
I've moved all the node specific stuff into this adapter function. You pass it an options hash and it returns a promise. This makes mocking the http level inetractions easier and will also enable us to easily switch out this adapter if we ever choose.

Reply:
Still have some refactoring to do on it

Reply:
I think requestBody could be on the options, but i'm trying to keep the options pure as only options that the node http module accepts.

Reply:
If you wanted to intermingle them, it would be straightforward to destructure them in the signature. The caller shouldn't care about the internal purity of the function, so no concern there. That said I think this impl is perfectly fine :]

```js
function makeRequest ({ requestBody, ...options }) {
```